components/identity/org.wso2.carbon.identity.certificateauthority/src/main/java/org/wso2/carbon/identity/certificateauthority/CAAdminService.java

```
@@ -0,0 +1,288 @@
+     *
+     * @param serialNo The serial number of the CSR
+     * @return CSR with the given serial number
+     * @throws CAException
+     */
+    public CSR getCSR(String serialNo) throws CAException {
+        try {
+            String tenantDomain = CarbonContext.getThreadLocalCarbonContext().getTenantDomain();
+            return csrService.getCSR(serialNo, tenantDomain);
+        } catch (CAException e) {
+            log.error("Error retrieving the CSR with serial no:" + serialNo, e);
+            throw new CAException("Error when retrieving the CSR");
+        }
+    }
+
+    /**
+     * Reject CSR without signing.
+     *
+     * @param serialNo The serial number of the CSR to be rejected
+     * @throws CAException
+     */
+    public void rejectCSR(String serialNo) throws CAException {
+        try {
+            String tenantDomain = CarbonContext.getThreadLocalCarbonContext().getTenantDomain();
+            csrService.rejectCSR(serialNo, tenantDomain);
+        } catch (CAException e) {
+            log.error("Error when rejecting the CSR with serial no:" + serialNo, e);
+            throw new CAException("Error when rejecting the CSR");
+        }
+    }
+
+    /**
+     * Signs the CSR and stores the resulting certificate.
+     *
+     * @param serialNo The serial number of the CSR to be signed
+     * @param validity The number of days that the resulting certificate should be valid before expiration
+     * @throws CAException
+     */
+    public void signCSR(String serialNo, int validity) throws CAException {
+        try {
+            String tenantDomain = CarbonContext.getThreadLocalCarbonContext().getTenantDomain();
+            certificateService.signCSR(tenantDomain, serialNo, validity);
+        } catch (CAException e) {
+            log.error("Error signing the CSR with serial no:" + serialNo + " , for " + validity + " days", e);
+            throw new CAException("Certificate could not be signed");
```

The original exception is swallowed here. Please fix.